src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Modules API: WP_Modules class.
+ *
+ * Native support for ES Modules and Import Maps.
+ *
+ * @package WordPress
+ * @subpackage Modules
+ */
+
+/**
+ * Core class used to register modules.
+ *
+ * @since 6.5.0
+ */
+class WP_Modules {
+	/**
+	 * Holds the registered modules, keyed by module identifier.
+	 *
+	 * @since 6.5.0
+	 * @var array
+	 */
+	private $registered = array();
+
+	/**
+	 * Holds the module identifiers that were enqueued before registered.
+	 *
+	 * @since 6.5.0
+	 * @var array
+	 */
+	private $enqueued_before_registered = array();
+
+	/**
+	 * Registers the module if no module with that module identifier has already
+	 * been registered.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @param string            $module_identifier The identifier of the module. Should be unique. It will be used in the
+	 *                                             final import map.
+	 * @param string            $src               Full URL of the module, or path of the script relative to the WordPress
+	 *                                             root directory.
+	 * @param array             $dependencies      Optional. An array of module identifiers of the dependencies of this
```

Comment:
Perhaps this could be more explicit? Granted, using array shapes in core is not common, although they are used in `src/wp-includes/sodium_compat`:

```suggestion
	 * @param array<string|array{id: string, type?: 'static'|'dynamic' }>             $dependencies      Optional. An array of module identifiers of the dependencies of this
```

Reply:
That's better, thanks. Done in https://github.com/WordPress/wordpress-develop/pull/5818/commits/df73485c9c43995eea8b298138df43e258cef0d4.